pkg/trace/test/testsuite/cards_test.go

```diff
@@ -36,27 +76,109 @@ apm_config:
   obfuscation:
     credit_cards:
       enabled: true`),
-			out: "?",
+			out:     "?",
+			version: "v0.6",
 		},
 	} {
-		t.Run(tt.out, func(t *testing.T) {
+		t.Run(tt.version+"/"+tt.out, func(t *testing.T) {
 			if err := r.RunAgent(tt.conf); err != nil {
 				t.Fatal(err)
 			}
 			defer r.KillAgent()
 
-			p := testutil.GeneratePayload(1, &testutil.TraceConfig{
-				MaxSpans: 1,
-				Keep:     true,
-			}, &testutil.SpanConfig{MinTags: 2})
-			p[0][0].Meta["credit_card_number"] = "4166 6766 6766 6746"
-			if err := r.Post(p); err != nil {
+			payload, traces := generatePayload(tt.version, "4166 6766 6766 6746")
+			if err := r.PostMsgpack("/"+tt.version+"/traces", payload); err != nil {
 				t.Fatal(err)
 			}
 			waitForTrace(t, &r, func(v pb.AgentPayload) {
-				payloadsEqual(t, p, v)
+				payloadsEqual(t, traces, v)
 				assert.Equal(t, v.TracerPayloads[0].Chunks[0].Spans[0].Meta["credit_card_number"], tt.out)
 			})
 		})
 	}
 }
+
+func generatePayload(version string, cardNumber string) (msgp.Marshaler, pb.Traces) {
```

Comment:
```suggestion
func generatePayload(version, cardNumber string) (msgp.Marshaler, pb.Traces) {
```

Reply:
I'll use `api.Version` instead of `string`, here.